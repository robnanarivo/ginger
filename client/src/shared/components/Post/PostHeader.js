import { React } from 'react';
import { useHistory } from 'react-router-dom';
import CardHeader from '@mui/material/CardHeader';
import Avatar from '@mui/material/Avatar';
import { red } from '@mui/material/colors';

const PostHeader = ({
  creatorId,
  creatorName,
  profilePicture,
  dateString,
}) => {
  const history = useHistory();

  const viewProfile = () => {
    history.push(`/user/${creatorId}`);
  };

  return (
    <CardHeader
      avatar={
        profilePicture
          ? (
            <Avatar src={profilePicture} sx={{ bgcolor: red[500], '&:hover': { cursor: 'pointer' } }} onClick={viewProfile} />
          )
          : (
            <Avatar sx={{ bgcolor: red[500], '&:hover': { cursor: 'pointer' } }} onClick={viewProfile}>
              {creatorName.charAt(0)}
            </Avatar>
          )
      }
      title={creatorName}
      subheader={dateString}
    />
  );
};

export default PostHeader;
