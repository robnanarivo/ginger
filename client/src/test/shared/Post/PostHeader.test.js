import React from 'react';
import renderer from 'react-test-renderer';
import PostHeader from '../../../shared/components/Post/PostHeader';
import 'jest-styled-components';

describe('PostHeader unit test', () => {
  it('PostHeader Snapshot test', () => {
    const tree = renderer
      .create(
        <PostHeader creatorName="Bob" />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
