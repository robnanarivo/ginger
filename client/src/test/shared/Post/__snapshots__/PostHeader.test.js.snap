// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostHeader unit test PostHeader Snapshot test 1`] = `
<div
  className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
>
  <div
    className="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-x7yxgc-MuiAvatar-root"
      onClick={[Function]}
    >
      B
    </div>
  </div>
  <div
    className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
  >
    <span
      className="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
    >
      Bob
    </span>
  </div>
</div>
`;
